Content.Shared/Atmos/Components/GasMinerComponent.cs

```diff
@@ -0,0 +1,47 @@
+using Content.Shared.Atmos;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Atmos.Components;
+
+[NetworkedComponent]
+[AutoGenerateComponentState]
+[RegisterComponent]
+public sealed partial class GasMinerComponent : Component
+{
+    [AutoNetworkedField]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public bool Enabled { get; set; } = true;
+
+    [AutoNetworkedField]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public bool Idle { get; set; } = false;
+
+    /// <summary>
+    ///      If the number of moles in the external environment exceeds this number, no gas will be mined.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("maxExternalAmount")]
+    public float MaxExternalAmount { get; set; } = float.PositiveInfinity;
```

Comment:
```suggestion
    [DataField]
    public float MaxExternalAmount = float.PositiveInfinity;
```
(do this for all of them)